src/qibolab/execution_parameters.py

```diff
@@ -87,7 +66,18 @@ class ExecutionParameters(Model):
     top of platform defaults.
     """
 
-    @property
-    def results_type(self):
-        """Returns corresponding results class."""
-        return RESULTS_TYPE[self.averaging_mode][self.acquisition_type]
+    def results_shape(
+        self, sweepers: list[ParallelSweepers], samples: Optional[int] = None
+    ) -> tuple[int, ...]:
+        """Compute the expected shape for collected data."""
+
+        shots = [self.nshots] if self.averaging_mode is AveragingMode.SINGLESHOT else []
+        sweeps = [
+            min(len(sweep.values) for sweep in parsweeps) for parsweeps in sweepers
+        ]
+        inner = {
+            AcquisitionType.DISCRIMINATION: [],
+            AcquisitionType.INTEGRATION: [2],
+            AcquisitionType.RAW: [samples, 2],
+        }[self.acquisition_type]
+        return tuple(shots + sweeps + inner)
```

Comment:
No special reason, just a couple of habits:
- the literal is more intuitive (in general also the tuple is simple, but there is no comprehension for tuples, and instead `()` are used for the generator - but you can use `tuple(x for x in ...)`)
- in general, tuples are much stricter than lists, since each tuple length correspond to a different type (even for homogeneous tuples) - but in Python they still support the concatenation (though creating a new object - which is fine here)

So, yes, I can convert it.

Reply:
Done in 1e214df131e69ab97f01c110b6c07da3e2b7086f